src/MacVim/MMWindowController.h

```diff
@@ -19,30 +19,32 @@
 
 @interface MMWindowController : NSWindowController<NSWindowDelegate>
 {
-    MMVimController     *vimController;
-    MMVimView           *vimView;
-    BOOL                setupDone;
-    BOOL                windowPresented;
-    BOOL                shouldResizeVimView;
-    BOOL                shouldRestoreUserTopLeft;
-    BOOL                shouldMaximizeWindow;
-    int                 updateToolbarFlag;
-    BOOL                keepOnScreen;
-    NSString            *windowAutosaveKey;
-    BOOL                fullScreenEnabled;
-    MMFullScreenWindow  *fullScreenWindow;
-    int                 fullScreenOptions;
-    BOOL                delayEnterFullScreen;
-    NSRect              preFullScreenFrame;
-    MMWindow            *decoratedWindow;
-    NSString            *lastSetTitle;
-    int                 userRows;
-    int                 userCols;
-    NSPoint             userTopLeft;
-    NSPoint             defaultTopLeft;
-    NSToolbar           *toolbar;
-    BOOL                resizingDueToMove;
-    int                 blurRadius;
+    MMVimController               *vimController;
+    MMVimView                     *vimView;
+    BOOL                          setupDone;
+    BOOL                          windowPresented;
+    BOOL                          shouldResizeVimView;
+    BOOL                          shouldRestoreUserTopLeft;
+    BOOL                          shouldMaximizeWindow;
+    int                           updateToolbarFlag;
+    BOOL                          keepOnScreen;
+    NSString                      *windowAutosaveKey;
+    BOOL                          fullScreenEnabled;
+    MMFullScreenWindow            *fullScreenWindow;
+    int                           fullScreenOptions;
+    BOOL                          delayEnterFullScreen;
+    NSRect                        preFullScreenFrame;
+    MMWindow                      *decoratedWindow;
+    NSString                      *lastSetTitle;
+    int                           userRows;
+    int                           userCols;
+    NSPoint                       userTopLeft;
+    NSPoint                       defaultTopLeft;
+    NSToolbar                     *toolbar;
+    BOOL                          resizingDueToMove;
+    int                           blurRadius;
+    NSMutableArray<NSOperation *> *afterWindowPresentedQueue;
+    NSLock                        *afterWindowPresentedQueueLock;
```

Comment:
I grepped^ and temporarily put some debug NSException conditions in for non-main thread and `trylock` not succeeding... seems like all main-thread with very anecdotal, meat-puppet CI.  `MMAppController` points to threading issues but not window controller / event processing related. If/when migrating to Swift, it would probably be worth considering rewriting event dispatching using Notification Center and GCD to prevent the UI from hanging (unless it's desirable) during long operations.

^ `egrep -ri '(queue|thread|event|selector|operation|dispatch)' src/MacVim | grep '^[A-Za-z0-9_+]*\.[mh]: ' | less`